Extending/Packages/Packages-on-Umbraco-Cloud/index.md

```diff
@@ -4,9 +4,42 @@ versionFrom: 8.0.0
 
 # Packages on Umbraco Cloud
 
-If you want to use or develop packages for Umbraco Cloud there are a few things to consider and be aware of. One such thing is that custom property editors may need a **ValueConnector** to transform their content between environments.
+If you want to use or develop packages for Umbraco Cloud there are a few things to consider and be aware of.
+The two most important things to know about are
+- [How you should store data on Cloud](#storing-data)
+- [Using custom property editors with Deploy](#valueconnectors)
 
-# ValueConnectors
+## Storing data
+
+When developing a package you will sometimes store data, this can be data in many forms - Umbraco schema / content, package settings, etc.
+
+When you develop a package for Umbraco Cloud there are a few things to be aware of when storing data, mainly whether you want that data to be specific to 1 environment or more.
+
+Let's take a look at the most common ways of storing data in packages - and what to watch out for on Cloud:
+
+### Package actions
+
+A [package action](../Package-Actions/custom-package-actions.md) are used to run some commands on package install and uninstall. One thing to watch out for with regards to package actions is that whatever code you run in this will only be executed on the environment you install the package on. So if you for example set up a database table in a package action that db table will not be generated in each environments db.
+If you need a db table in each environments db you will probably need to look at migrations below instead.
+
+### Migrations
+
+A [database migration](../../Database/index.md) is a set of database commands you run as part of a migration plan. That migration plan has an id that is stored in the database (in the KeyValue table). This means that when you add new migrations Umbraco will only execute the ones that came after the one with the stored id.
+
+As database migrations are run in the database of the site it also means that they will run on each environment you trigger them on. The most common way to trigger a db migration is to include them in a [composer](../../../Implementation/Composing/index.md), which will ensure they run on site startup. This means any database commands you have in your migration will automatically run when the site starts up. When your package code is pushed to a new environment it will run them from the beginning on that environment as no id is saved in the database.
```

Comment:
```suggestion
As database migrations are run in the database of the site it also means that they will run on each environment you trigger them on. The most common way to trigger a database migration is to include them in a [composer](../../../Implementation/Composing/index.md), which will ensure they run on site startup. This means that any database commands you have in your migration will automatically run when the site starts up. When your package is pushed to a new environment it will run the code from the beginning on that environment as no ID is saved in the database.
```